### PR TITLE
Updated collection types to modern objc syntax.

### PR DESCRIPTION
**Pull request by namalu**

### CODE REVIEW

**Review by mikenevi — APPROVED** [review]

File: MHVLib/Types/MHVItem.m [hunk 1]
```diff
@@ -396,144 +370,153 @@ -(void)prepareForUpdate
...
 {
-    return ([self indexOfItemID:itemID] != NSNotFound);
+    return [self indexOfItemID:itemID] != NSNotFound;
 }
 
--(NSUInteger)indexOfItemID:(NSString *)itemID
+- (NSUInteger)indexOfItemID:(NSString *)itemID
 {
-    for (NSUInteger i = 0, count = m_inner.count; i < count; ++i)
+    for (NSUInteger i = 0, count = self.count; i < count; ++i)
```
Comment (miburf) [inline MHVLib/Types/MHVItem.m]: nit, can remove count= part and do i < self.count

File: MHVLib/Types/MHVItem.m [hunk 2]
```diff
@@ -543,135 +526,137 @@ -(NSUInteger)indexOfItemID:(NSString *)itemID
...
+- (NSMutableDictionary *)getItemsIndexedByID
 {
     return [self newIndexByID];
 }
 
--(NSUInteger)indexOfTypeID:(NSString *)typeID
+- (NSUInteger)indexOfTypeID:(NSString *)typeID
 {
-    for (NSUInteger i = 0, count = m_inner.count; i < count; ++i)
+    for (NSUInteger i = 0, count = self.count; i < count; ++i)
```
Comment (miburf) [inline MHVLib/Types/MHVItem.m]: here too, likely are more we can find by searching for ", count ="

File: MHVLib/Types/MHVItem.m [hunk 3]
```diff
@@ -543,135 +526,137 @@ -(NSUInteger)indexOfItemID:(NSString *)itemID
...
     }
 
-    return TRUE;
-    
-LError:
-    return FALSE;
+    return YES;
+
+   LError:
+    return NO;
```
Comment (miburf) [inline MHVLib/Types/MHVItem.m]: can delete

File: MHVLib/Types/MHVRecord.m [hunk 4]
```diff
@@ -27,131 +27,134 @@
...
 
 @end
 
+@interface MHVRecordCollection ()
+
+@property (nonatomic, strong) NSMutableArray *inner;
+
+@end
+
 @implementation MHVRecordCollection
```
Comment (miburf) [inline MHVLib/Types/MHVRecord.m:112]: I think inner need to be @dynamic since MHVCollection has inner too.

**Review by miburf — APPROVED** [review]

File: MHVLib/Types/MHVRecord.m (hunk 4 above)
Comment (namalu) [reply MHVLib/Types/MHVRecord.m:112]: I don't think it matters because inner is not "public" in the super.

**Pull request merged**